script.js

```diff
@@ -1,4 +1,446 @@
+        cardName = "King";
+      }
+
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        suitPic: currentSuitPic,
+        rank: rankCounter,
+      };
+
+      cardDeck.push(card);
+
+      rankCounter += 1;
+    }
+    suitIndex += 1;
+  }
+
+  return cardDeck;
+};
+var getRandomIndex = function (size) {
+  var randomInteger = Math.floor(Math.random() * size);
+  return randomInteger;
+};
+var shuffleCards = function () {
+  var cardDeck = makeDeck();
+  numberOfCards = cardDeck.length;
+
+  var currentIndex = 0;
+
+  while (currentIndex < numberOfCards) {
+    var randomIndex = getRandomIndex(numberOfCards);
+    var randomCard = cardDeck[randomIndex];
+    var currentCard = cardDeck[currentIndex];
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+
+    currentIndex = currentIndex + 1;
+  }
+
+  return cardDeck;
+};
+
+//GAME MODES
+// Enter number of players
+var GAME_MODE_ENTER_NUMBER_OF_PLAYERS = "ENTER_NUMBER_OF_PLAYERS";
+// Enter Name: Output - Name & Click submit to deal cards.
+// Input = name. Output: `Hi ${input}, click "submit" to deal cards.`
+var GAME_MODE_ENTER_NAME = "ENTER_NAME";
+//Betting
+// 1. Player starts with 100 points. Each round the player wagers a number of points before their hand is dealt. Lower point = better
+var GAME_MODE_WAGER = "WAGER";
+// Deal Cards: Output - Details of cards & Type "hit"/"submit"
+// Input: Nothing. Output: The cards are dealed. Hi player, click "submit" to view your cards.
+var GAME_MODE_DEAL_CARDS = "DEAL_CARDS";
+// Decides: "hit" or "stand"
+var GAME_MODE_DECISION = "DECISION";
+var GAME_MODE_POST_DECISION = "POST_DECISION";
+// If user hits "stand" & computer makes sense to stand:  -- calculate result of both & shows the result of both
+var GAME_MODE_LEADERBOARD = "LEADERBOARD";
+// ACE
+var GAME_MODE_ACE_POST_DECISION = "ACE_POST_DECISION";
+
+// Current game mode
+var gameMode = GAME_MODE_ENTER_NUMBER_OF_PLAYERS;
+var currentPlayer = 1;
+var numOfPlayers = 0;
+var wager = 0;
+var userName = [];
+var cardNumberPerPlayer = []; // player's card number
+var numberofWinsPerPlayer = [];
+var numberOfGamesPerPlayer = [];
+var wagerPointSystem = [];
+var cardsPerRound = [];
+var playerCardsTemp = [];
+var newDeck = [];
+
+var indexOfMaxInArray = function (cardNumberPerPlayer) {
+  if (cardNumberPerPlayer.length == 0) {
+    return -1;
+  }
+
+  var max = cardNumberPerPlayer[0];
+  var maxIndex = 0;
+
+  for (var i = 1; i < cardNumberPerPlayer.length; i++) {
+    if (cardNumberPerPlayer[i] > max) {
+      maxIndex = i;
+      max = cardNumberPerPlayer[i];
+    }
+  }
+
+  return maxIndex;
+};
+var incrementNumberOfWinsOfEachPlayer = function (returnIndexOfHighestNumber) {
+  numberofWinsPerPlayer[returnIndexOfHighestNumber] += 1;
+};
+var countWagerPointSystem = function (returnIndexOfHighestNumber) {
+  wagerPointSystem[returnIndexOfHighestNumber] -= wager;
+};
+
 var main = function (input) {
```

Comment:
great job being able to implement the game logic here. A point of improvement would be to break up your game logic into smaller functions. For example, if we look at your main function here, it is nearly 300 lines, which is too long for any function. Ideally, our function should only do one thing, and not everything at once. For example, this main function should only tell me how it routes different game modes to different helper functions, and helper functions itself can tell me the game logic to see who wins